src/events/eventBus/EventBusHandler.ts

```diff
@@ -0,0 +1,108 @@
+ * handler function is asynchronously sent to the Event Bus described
+ * in the application config file.
+ *
+ * To handle a new event, you need not update this class directly. Instead,
+ * add a payload builder function to EventBusConfig.ts and update the
+ * mapping of event name to payload builder in that file.
+ *
+ * Error Handling:
+ *  Any error that occurs when either building the payload or sending the
+ *  event to EventBridge will be sent to Sentry and logged in Cloudwatch.
+ *  No errors are thrown directly.
+ *
+ **/
+export class EventBusHandler {
+  // Mapping of internal application events to listeners in this class
+  // Update this in constructor whenever new listener methods are added
+  // The key is optional since not all possible events delineated in the
+  // app logic are consumed
+  private eventHandlerCallbackMap: {
+    [key in BaseEventData['eventType']]?: (data: any) => BaseEventBusPayload;
+  };
+
+  private client: EventBridgeClient;
+
+  /**
+   * When an instance of EventBusHandler is constructed, register listeners
+   * on the passed `emitter` object according to the specification in
+   * `eventHandlerMap`
+   * @param emitter the emitter which sends events to listen to
+   * @param eventHandlerMap optional map of event name to payload builder
+   * functions. If not specified, will default to eventBusConfig
+   */
+  constructor(
+    protected emitter: EventEmitter,
+    eventHandlerMap?: EventHandlerCallbackMap
+  ) {
+    this.client = new EventBridgeClient({
+      region: config.aws.region,
+    });
+    // Update this mapping with the new handler function and event whenever
+    // new events/handler methods are added
+    this.eventHandlerCallbackMap = eventHandlerMap ?? eventBusConfig;
+    // Register handler method for each handled event
+    Object.entries(this.eventHandlerCallbackMap).forEach(([event, method]) => {
+      emitter.on(event, (data) => {
+        try {
+          const eventPayload = method(data);
+          this.sendEvent(eventPayload);
+        } catch (error) {
+          // In the unlikely event that the payload generator throws an error,
+          // log to Sentry and Cloudwatch but don't halt program
+          Sentry.captureException(error);
+          console.log(error);
+        }
+      });
+    });
+  }
+
+  /**
+   * Send event to Event Bus, pulling the event bus and the event source
+   * from the config.
+   * Will not throw errors if event fails; instead, log exception to Sentry
+   * and add to Cloudwatch logs.
+   * @param eventPayload the payload to send to event bus
+   */
+  async sendEvent(eventPayload: BaseEventBusPayload) {
+    const putEventCommand = new PutEventsCommand({
+      Entries: [
+        {
+          EventBusName: config.aws.eventBus.name,
+          Detail: JSON.stringify(eventPayload),
+          Source: config.eventBridge.source,
+          DetailType: config.eventBridge.addScheduledItemEventType,
```

Comment:
This will always send the same `detail-type`. We need to send different `detail-type`s based on the action.
Ex. `update-scheduled-item` and `delete-scheduled-item`

Reply:
good catch, I forgot to update that

Reply:
it's done